File: problemas/matematica/2497/2497.ml
let rec leitura (teste: int) =
  try
    Scanf.scanf "%d\n" (fun n ->
        if n != -1 then (
					Printf.printf "Experiment %d: %d full cycle(s)\n" teste (n / 2);
					leitura (teste + 1)
		))
  with End_of_file -> ()

let () = leitura 1

File: problemas/matematica/2497/2497.c
#include <stdio.h>

int main(){
    int n, teste;

    teste = 1;
    while (1) {
        scanf("%d\n", &n);

        if (n == -1) {
            break;
        }

        printf("Experiment %d: %d full cycle(s)\n", teste, n / 2);
        teste += 1;
    }

    return 0;
}